proyecto-frontend/src/app/modules/product/component/category/category.component.ts

```diff
@@ -1,24 +1,161 @@
+  showAllCategories: boolean = false;
+
+  constructor(private service: CategoryService, private fb: FormBuilder) {
+    this.categoryForm = this.fb.group({
+      category: ['', Validators.required],
+      tag: ['', Validators.required],
+      status: ['', Validators.required],
+    });
+  }
 
-  constructor(private service: CategoryService) {}
+  successMessage(message: string) {
+    this.swal.fire({
+      icon: 'success',
+      title: '¡Éxito!',
+      text: message,
+    });
+  }
+
+  errorMessage(message: string) {
+    this.swal.fire({
+      icon: 'error',
+      title: 'Error',
+      text: message,
+    });
+  }
 
   getCategories(): void {
-    this.categories = this.service.getCategories();
+    this.service.getCategories().subscribe({
+      next: (v) => {
+        this.isActiveList = false;
+        this.categories = v;
+      },
+      error: (e) => {
+        console.log(e);
+      },
+    });
+  }
+  toggleCategoryFilter(): void {
+    this.showAllCategories = !this.showAllCategories;
+    if (this.showAllCategories) {
+      this.getCategories();
+    } else {
+      this.getActiveCategories();
+    }
+  }
+
+  getActiveCategories(): void {
+    this.service.getActiveCategories().subscribe({
+      next: (v) => {
+        this.isActiveList = true;
+        this.categories = v;
+      },
+      error: (e) => {
+        console.log(e);
+      },
+    });
   }
 
   ngOnInit(): void {
     this.getCategories();
   }
+
+  onSubmit(): void {
+    if (this.category_id === 0) {
+      this.onSubmitCreate();
+    } else {
+      this.onSubmitUpdate();
+    }
+
+    this.category_id = 0;
+  }
+
+  onSubmitCreate(): void {
+    this.service.createCategory(this.categoryForm.value).subscribe({
+      next: (v) => {
+        this.resetCategories();
+        this.categoryForm.reset();
+        this.successMessage(v.message);
+      },
+      error: (e) => {
+        this.errorMessage(e.error.message);
+      },
+    });
+  }
+
+  onSubmitUpdate() {
+    this.service
+      .updateCategory(this.categoryForm.value, this.category_id)
+      .subscribe({
+        next: (v) => {
+          this.resetCategories();
+          this.categoryForm.reset();
+          this.successMessage(v.message);
+        },
+        error: (e) => {
+          this.errorMessage(e.error.message);
+        },
+      });
+  }
+
+  disableCategory(id: number) {
+    this.service.deleteCategory(id).subscribe({
+      next: (v) => {
+        this.resetCategories();
+        this.successMessage(v.message);
+      },
+      error: (e) => {
+        this.errorMessage(e.error!.message);
+      },
+    });
+  }
+
+  enableCategory(id: number) {
+    this.service.activateCategory(id).subscribe({
+      next: (v) => {
+        this.resetCategories();
+        this.successMessage(v.message);
+      },
+      error: (e) => {
+        this.errorMessage(e.error!.message);
+      },
+    });
+  }
+
+  updateCategory(id: number) {
+    this.category_id = id;
```

Comment:
Recuerda que debemos reiniciar el modal siempre que lo abrimos.
Además, cuando actualizamos un elemento, usualmente queremos observar lo que vamos a editar.

Esto se puede hacer de la siguiente manera:
```suggestion
    this.category_id = id;
    
    this.categoryForm.reset();
    this.categoryForm.controls['category'].setValue(Category.category);
    this.categoryForm.controls['tag'].setValue(Category.tag);        
```